apps/emqx_sn/src/emqx_sn_gateway.erl

```diff
@@ -590,11 +587,7 @@ handle_call(_From, Req, State = #state{channel = Channel}) ->
         {reply, Reply, NChannel} ->
             {reply, Reply, State#state{channel = NChannel}};
         {shutdown, Reason, Reply, NChannel} ->
-            shutdown(Reason, Reply, State#state{channel = NChannel});
-        {shutdown, Reason, Reply, OutPacket, NChannel} ->
```

Comment:
This case clause should be kept here?

In the `emqx_channel:handle_call/2`, Its spec has pointed out this type:
```
-spec(handle_call(Req :: term(), channel())
      -> {reply, Reply :: term(), channel()}
       | {shutdown, Reason :: term(), Reply :: term(), channel()}
       | {shutdown, Reason :: term(), Reply :: term(), emqx_types:packet(), channel()}).
```

Reply:
According to Dialyzer, OutPacket contains either #mqtt_packet_auth{} or #mqtt_packet_disconnect{}. mqtt2sn/2, however, does not have such clauses to handle those two type of packets. That is why I assume this clause is unused.